### PR TITLE
Make the default open program be based on the OS.

### PR DESCRIPTION
**Pull request by chick**
Use open for mac, xdg-open for linux.
For the moment it is not clear what to use on windows or anything else
so there will be no default there.

### CODE REVIEW

File: src/main/scala/dotvisualizer/FirrtlDiagrammer.scala [hunk 1]
```diff
@@ -255,3 +255,18 @@ case class Config(
...
+object Config {
+  private val MacPattern = """.*mac.*""".r
+  private val LinuxPattern = """.*n[iu]x.*""".r
+  private val WindowsPattern = """.*win.*""".r
+
+  def getOpenForOs: String = {
+    System.getProperty("os.name").toLowerCase match {
+      case MacPattern()     => "open"
+      case LinuxPattern()   => "xdg-open"
+      case WindowsPattern() => ""          // no clear agreement here.
```
Comment (edwardcwang) [inline src/main/scala/dotvisualizer/FirrtlDiagrammer.scala]: Should we use `???` instead of "" here?

Comment (chick) [reply src/main/scala/dotvisualizer/FirrtlDiagrammer.scala]: I don't think so. I want it to run successfully here by default.  I should probably document more how to view if there is no open program

**Comment by chick** [comment]
@edwardcwang got rid of default "open"
added better message for when open is not set.

File: src/main/scala/dotvisualizer/FirrtlDiagrammer.scala [hunk 2]
```diff
@@ -65,11 +65,16 @@ object FirrtlDiagrammer {
     * @param fileName    file to be opened
     * @param openProgram program to use
     */
-  def show(fileName: String, openProgram: String = "open"): Unit = {
+  def show(fileName: String, openProgram: String): Unit = {
     if(openProgram.nonEmpty && openProgram != "none") {
```
[the remark this answers is not in the document]
Comment (chick) [reply src/main/scala/dotvisualizer/FirrtlDiagrammer.scala]: before when the default was open, it was tricky to get the ./diagrammer script to turn off the open.  You had to do -o '""' or something, it was easier to just do -o none. That's the history of it. I could pull it but it seems harmless

[the remark this answers is not in the document]
Comment (edwardcwang) [reply src/main/scala/dotvisualizer/FirrtlDiagrammer.scala]: Sounds good. Maybe add the explanation above as a comment? Otherwise LGTM.

**Comment by chick** [comment]
@edwardcwang can you approve this PR

**Comment by edwardcwang** [comment]
@chick Could you grant write access to do so?

**Comment by chick** [comment]
@edwardcwang I've added chisel devs write access, that should do it, let me know if there's a problem

**Pull request merged**